addons/project/security/project_security.xml

```diff
@@ -312,5 +312,19 @@
         <field name="domain_force">[(1, '=', 1)]</field>
         <field name="groups" eval="[(4, ref('project.group_project_manager'))]"/>
     </record>
+
+    <record id="project_milestone_rule_portal_project_sharing" model="ir.rule">
+        <field name="name">Project/milestone portal users: portal user can read with project sharing feature</field>
+        <field name="model_id" ref="project.model_project_milestone"/>
+        <field name="domain_force">[
+            ('project_id.privacy_visibility', '=', 'portal'),
+            ('project_id.collaborator_ids.partner_id', 'in', [user.partner_id.id]),
+        ]</field>
+        <field name="groups" eval="[(4, ref('base.group_portal'))]"/>
+        <field name="perm_read" eval="True"/>
+        <field name="perm_write" eval="False"/>
+        <field name="perm_create" eval="False"/>
+        <field name="perm_unlink" eval="False"/>
```

Comment:
We can let True by default 🙂 

Why? to be able to use that ir.rule if someone allows the portal user to create/write/delete by altering the ir.model.access. 🙂 

```suggestion
        <field name="perm_write" eval="True"/>
        <field name="perm_create" eval="True"/>
        <field name="perm_unlink" eval="True"/>
```